newIDE/app/src/ObjectGroupsList/index.js

```diff
@@ -67,430 +80,521 @@ type Props = {|
+        const { group, global } = groupWithContext;
+
+        const newName = newNameGenerator(
+          group.getName(),
+          name => objectGroups.has(name) || globalObjectGroups.has(name),
+          ''
+        );
+
+        const container: gdObjectGroupsContainer = global
+          ? globalObjectGroups
+          : objectGroups;
+
+        const serializedDuplicatedGroup = serializeToJSObject(group);
+        const newGroup = container.insertNew(
+          newName,
+          container.getPosition(group.getName()) + 1
+        );
+
+        unserializeFromJSObject(
+          newGroup,
+          serializedDuplicatedGroup,
+          'unserializeFrom'
+        );
+        newGroup.setName(newName); // Unserialization has overwritten the name.
+
+        onObjectGroupModified();
+      },
+      [globalObjectGroups, objectGroups, onObjectGroupModified]
     );
-    newGroup.setName(newName); // Unserialization has overwritten the name.
 
-    this._onObjectGroupModified();
-  };
+    const onRename = React.useCallback(
+      (groupWithContext: GroupWithContext, newName: string) => {
+        const { group, global } = groupWithContext;
+
+        if (group.getName() === newName) return;
 
-  _onRename = (groupWithContext: GroupWithContext, newName: string) => {
-    const { group, global } = groupWithContext;
+        const validatedNewName = getValidatedObjectOrGroupName(newName, global);
+        onRenameGroup(groupWithContext, validatedNewName, doRename => {
+          if (!doRename) return;
 
-    this.setState({
-      renamedGroupWithContext: null,
-    });
+          group.setName(validatedNewName);
 
-    if (group.getName() === newName) return;
+          onObjectGroupModified();
+          if (onGroupRenamed) {
+            onGroupRenamed();
+          }
+        });
+      },
+      [
+        getValidatedObjectOrGroupName,
+        onObjectGroupModified,
+        onRenameGroup,
+        onGroupRenamed,
+      ]
+    );
 
-    const validatedNewName = this.props.getValidatedObjectOrGroupName(
-      newName,
-      global
+    const setAsGlobalGroup = React.useCallback(
+      async (groupWithContext: GroupWithContext, index?: number) => {
+        const { group } = groupWithContext;
+
+        const groupName = group.getName();
+
+        if (globalObjectGroups.has(groupName)) {
+          await showAlert({
+            title: t`Set as global group`,
+            message: t`A global object with this name already exists. Please change the group name before setting it as a global group`,
+          });
+          return;
+        }
+
+        if (beforeSetAsGlobalGroup && !beforeSetAsGlobalGroup(groupName)) {
+          return;
+        }
+
+        const answer = await showConfirmation({
+          title: t`Set as global group`,
+          message: t`This group will be loaded and available in all the scenes. This is only recommended for groups that you reuse a lot and can't be undone. Make this group global?`,
```

Comment:
not sure I understand the sentence `this group will be loaded`? 
OK now I do, but it took me a while, should this be more user-friendly like `This group will appear in all scenes` or something to say it may impact performance, but not say that it's the engine's fault 😅 ?

Reply:
That's a good question, I did not change the copy, but it sure could be improved. @LuniMoon Do you a suggestion?

Reply:
Suggestion:
Title: Get as global group
Message: Global elements help to manage [common? / prevalent?] objects across multiple scenes and it is recommended for the most used objects. This action cannot be undone. <break>
Do you want to set as global group? 
Set as global / Cancel